From fb718b793434b23b70f796b425b98a5aff4f871d Mon Sep 17 00:00:00 2001
From: Andrey Lazarev <revenue7@gmail.com>
Date: Wed, 26 Aug 2026 09:34:33 +0300
Subject: [PATCH 3/4] arm64: dts: rk3588s-rock-5a: add TPM 2.0 on spi4 for
 U-Boot

Enable spi4 (0xfecb0000) with the SPI4_M2 pin group and describe the Infineon SLB9670 TPM 2.0 module on it. The chip select is driven as a GPIO (GPIO1_A3, the same pad as SPI4_CS0_M2): the Rockchip SPI driver in U-Boot cannot hold the native chip select across the multi-part transfers that TPM TIS requires.

Upstream-Status: Inappropriate [board-specific hardware configuration]
---
 arch/arm/dts/rk3588s-rock-5a-u-boot.dtsi | 14 ++++++++++++++
 1 file changed, 14 insertions(+)

diff --git a/arch/arm/dts/rk3588s-rock-5a-u-boot.dtsi b/arch/arm/dts/rk3588s-rock-5a-u-boot.dtsi
index a8617333..1f36bc28 100644
--- a/arch/arm/dts/rk3588s-rock-5a-u-boot.dtsi
+++ b/arch/arm/dts/rk3588s-rock-5a-u-boot.dtsi
@@ -63,6 +63,20 @@
 	mmc-hs200-1_8v;
 };
 
+&spi4 {
+	status = "okay";
+	pinctrl-names = "default";
+	pinctrl-0 = <&spi4m2_pins>;
+	cs-gpios = <&gpio1 RK_PA3 GPIO_ACTIVE_LOW>;
+	num-cs = <1>;
+
+	tpm@0 {
+		compatible = "tcg,tpm_tis-spi", "infineon,slb9670";
+		reg = <0>;
+		spi-max-frequency = <500000>;
+	};
+};
+
 &u2phy0 {
 	status = "okay";
 };
-- 
2.43.0

